core/main/src/firebolt/firebolt_gateway.rs

```diff
@@ -146,6 +149,61 @@ impl FireboltGateway {
         }
     }
 
+    fn handle_broker_callback(
+        platform_state: PlatformState,
+        rpc_request: RpcRequest,
+    ) -> Sender<BrokerOutput> {
+        const REQUESTOR_CALLBACK_TIMEOUT_SECS: u64 = 10;
+
+        let (requestor_callback_tx, mut requestor_callback_rx) =
+            tokio::sync::mpsc::channel::<BrokerOutput>(1);
+
+        let start = Utc::now().timestamp_millis();
+        let protocol = rpc_request.ctx.protocol.clone();
+
+        tokio::spawn(async move {
+            let resp = tokio::time::timeout(
+                std::time::Duration::from_secs(REQUESTOR_CALLBACK_TIMEOUT_SECS),
+                requestor_callback_rx.recv(),
+            )
+            .await;
+
+            trace!("handle_broker_callback: resp={:?}", resp);
+
+            match resp {
+                Ok(has_broker_output) => {
+                    if let Some(broker_output) = has_broker_output {
+                        let now = Utc::now().timestamp_millis();
+
+                        let mut api_message = ApiMessage::new(
+                            protocol,
+                            serde_json::to_string(&broker_output.data).unwrap_or_default(),
```

Comment:
Is it better to handle the errors here and log those instead of going ahead with some default values ?

Reply:
Added, thanks.